src/java.base/share/classes/java/lang/runtime/ValueObjectMethods.java

```diff
@@ -662,13 +694,9 @@ private static LinkageError newLinkageError(Throwable e) {
      * <li>If {@code a} and {@code b} are both values of the same builtin primitive type,
      *     this method returns {@code a == b} with the following exception:
      *     <ul>
-     *     <li> If {@code a} and {@code b} both represent {@code NaN},
-     *          this method returns {@code true}, even though {@code NaN == NaN}
-     *          has the value {@code false}.
-     *     <li> If {@code a} is floating point positive zero while {@code b} is
-     *          floating point negative zero, or vice versa, this method
-     *          returns {@code false}, even though {@code +0.0 == -0.0} has
-     *          the value {@code true}.
+     *     <li> For primitive types {@code float} and {@code double} the
```

Comment:
I'm not sure if this is public documentation for user.

If it is then it might be helpful with a description of that the new equality semantics actually implies for NaN and the various zeros and infinities.

Having to look up `toRawBits` and figure it out will be hard for most people.

Reply:
These are implementation notes; the JLS specifies the behavior of value classes, operations of values, and the details of what it means for a value to be substitutable. Its still open as to how/where the semantics would appear in javadoc.